src/models/util/ParseCalChart31.ts

```diff
@@ -0,0 +1,355 @@
+    offset = this.ParseSHETREFPs(stuntSheet, buffer, offset);
+    offset = this.ParseSHETSYMBs(stuntSheet, buffer, offset);
+    offset = this.ParseSHETTYPEs(stuntSheet, buffer, offset);
+    offset = this.ParseSHETLABLs(stuntSheet, buffer, offset);
+    offset = this.ParseSHETCONTs(stuntSheet, buffer, offset);
+
+    if (readFourCharCode(buffer, offset) !== 'END ') {
+      throw 'Did not find the sheet END.';
+    }
+    offset += 4;
+    if (readFourCharCode(buffer, offset) !== 'SHET') {
+      throw 'Did not find the sheet END.';
+    }
+    offset += 4;
+
+    show.stuntSheets.push(stuntSheet);
+    return offset;
+  }
+
+  ParseSHETNAME(stuntSheet: StuntSheet, buffer: DataView, offset: number):
+  number {
+    const nextSection = readFourCharCode(buffer, offset);
+    if (nextSection !== 'NAME') {
+      throw 'Not a valid CalChart3.3.5 show.  Sheet missing name.';
+    }
+    offset += 4;
+    const size = readInt32(buffer, offset);
+    offset += 4;
+    const block = new DataView(
+      buffer.buffer,
+      buffer.byteOffset + offset,
+      size
+    );
+    stuntSheet.title = readStringTillEnd(block, 0);
+    return offset + size;
+  }
+
+  ParseSHETDURA(stuntSheet: StuntSheet, buffer: DataView, offset: number):
+  number {
+    if (readFourCharCode(buffer, offset) !== 'DURA') {
+      throw 'Not a valid CalChart3.3.5 show.  Sheet missing Duration.';
+    }
+    offset += 4;
+    if (readInt32(buffer, offset) !== 4) {
+      throw 'Not a valid CalChart3.3.5 show';
+    }
+    offset += 4;
+    stuntSheet.beats = readInt32(buffer, offset);
+    return offset + 4;
+  }
+
+  ParseSHETPOS(stuntSheet: StuntSheet, buffer: DataView, offset: number):
+  number {
+    if (readFourCharCode(buffer, offset) !== 'POS ') {
+      throw 'Not a valid CalChart3.3.5 show.  Sheet missing Duration.';
+    }
+    offset += 4;
+    let numPoints = readInt32(buffer, offset)/4;
+    if (numPoints !== this.numberDots) {
+      throw `POS error.  Expecting ${this.numberDots}, received ${numPoints}.`;
+    }
+    offset += 4;
+    const dots: StuntSheetDot[] = [];
+
+    // keep parsing points till data is exhausted
+    while (numPoints > 0) {
+      dots.push(new StuntSheetDot({
+        x: calChart3To4ConvertX(readInt16(buffer, offset)),
+        y: calChart3To4ConvertY(readInt16(buffer, offset + 2)),
+      }));
+      offset += 4;
+      --numPoints;
+    }
+    stuntSheet.stuntSheetDots = dots;
+    return offset;
+  }
+
+  ParseSHETREFPs(stuntSheet: StuntSheet, buffer: DataView, offset: number):
+  number {
+    while (readFourCharCode(buffer, offset) === 'REFP') {
+      offset = this.ParseSHETREFP(stuntSheet, buffer, offset);
+    }
+    return offset;
+  }
+
+  ParseSHETREFP(stuntSheet: StuntSheet, buffer: DataView, offset: number):
+  number {
+    if (readFourCharCode(buffer, offset) !== 'REFP') {
+      return offset;
+    }
+    offset += 4;
+    // references points not implemented.  Just skip for now
```

Comment:
For missing functionality, can you make an issue for them and comment the issue number? For example, "Implement reference points in Calchart 3 loaders". Each issue would be blocked until we implement them in Calchart 4.

Reply:
Done.